src/mmc_windows.c

```diff
@@ -33,163 +33,162 @@ void mmc_finalize()
 {
 }
 
-static int mmc_query_volume_path(const WCHAR *volume_path, struct mmc_device *device)
+/**
+ * @brief Scan for SDCards and other removable media
+ * @param devices where to store detected devices and some metadata
+ * @param max_devices the max to return
+ * @return the number of devices found
+ */
+int mmc_scan_for_devices(struct mmc_device *devices, int max_devices)
 {
-    HANDLE volume_handle = CreateFile(volume_path,
+    // There's not an API to request all the PhysicalDrives, but they are
+    // sequentially enumerated and become invalid if they're removed.
+    // So we just scan the first 256 of them until we find enough matches
+    int device_count = 0;
+    for (int i = 0; i < 256 && device_count < max_devices ; i++) {
+        WCHAR drive_path[MAX_PATH] = L"";
+        wsprintf(drive_path, L"\\\\.\\PhysicalDrive%d", i);
+
+        HANDLE drive_handle;
+        drive_handle = CreateFile(drive_path,
+                                  GENERIC_READ | GENERIC_WRITE,
+                                  FILE_SHARE_READ | FILE_SHARE_WRITE,
+                                  NULL,
+                                  OPEN_EXISTING,
+                                  0,
+                                  NULL);
+
+        if (drive_handle == INVALID_HANDLE_VALUE) {
+            CloseHandle(drive_handle);
+            continue;
+        }
+
+        DWORD bytes_returned;
+        DISK_GEOMETRY_EX geometry;
+        BOOL status = DeviceIoControl(drive_handle,
+                                      IOCTL_DISK_GET_DRIVE_GEOMETRY_EX,
+                                      NULL,
+                                      0,
+                                      &geometry,
+                                      sizeof(geometry),
+                                      &bytes_returned,
+                                      NULL);
+        CloseHandle(drive_handle);
+
+        if (status && geometry.Geometry.MediaType == RemovableMedia && geometry.DiskSize.QuadPart > 0) {
+            struct mmc_device *device;
+            device = &devices[device_count];
+            WideCharToMultiByte(CP_UTF8, 0, drive_path, -1, device->path, sizeof(device->path), NULL, NULL);
+            device->size = geometry.DiskSize.QuadPart;
+            device_count++;
+        }
+    }
+
+    return device_count;
+}
+
+/*
+ * @brief Query the Physical Drive(s) that back a Logical Volume
+ * Since we're trying dealing with SD cards, ignore any
+ * volumes that span multiple Physical Drives.
+ * @param volume_name the name of the volume as a Wide String
+ * @return the PhysicalDrive Number on success, 0 on failure
+ */
+unsigned int query_physical_extents(LPWSTR volume_name) {
```

Comment:
Make `query_physical_extents` a static function since it's not used outside of this file.